docs/_instructions/namesection.md

```diff
@@ -5,78 +5,73 @@ title: -namesection RESOURCE-SPEC ( ',' RESOURCE-SPEC ) *
 summary:  Create a name section (second part of manifest) with optional property expansion and addition of custom attributes. 
 ---
 
-	/**
-	 * Parse the namesection as instructions and then match them against the
-	 * current set of resources For example:
-	 * 
-	 * <pre>
-	 * 	-namesection: *;baz=true, abc/def/bar/X.class=3
-	 * </pre>
-	 * 
-	 * The raw value of {@link Constants#NAMESECTION} is used but the values of
-	 * the attributes are replaced where @ is set to the resource name. This
-	 * allows macro to operate on the resource
-	 */
+Create a name section (second part of manifest) with optional property expansion and addition of custom attributes.
 
-	private void doNamesection(Jar dot, Manifest manifest) {
+### Matching
+The key of the `-namesection` instruction is an _ant style_ glob. And there are two target groups for matching:
 
-		Parameters namesection = parseHeader(getProperties().getProperty(NAMESECTION));
-		Instructions instructions = new Instructions(namesection);
-		Set<String> resources = new HashSet<String>(dot.getResources().keySet());
+* **resources** - the pattern ends with `*` or is an exact match for a resource path
+* **packages** - the pattern ends with `/` or is an exact match for a package path 
 
-		//
-		// For each instruction, iterator over the resources and filter
-		// them. If a resource matches, it must be removed even if the
-		// instruction is negative. If positive, add a name section
-		// to the manifest for the given resource name. Then add all
-		// attributes from the instruction to that name section.
-		//
-		for (Map.Entry<Instruction,Attrs> instr : instructions.entrySet()) {
-			boolean matched = false;
+#### Custom attributes
 
-			// For each instruction
+The goal of named sections is to provide attributes over a specific subset of resources and paths in the jar file. Attributes are specified using the same syntax used elsewhere (such as package attributes). Attributes can contain properties and macros for expansion and replacement.
 
-			for (Iterator<String> i = resources.iterator(); i.hasNext();) {
-				String path = i.next();
-				// For each resource
+Each attribute is processed by bnd and the matching value is passed using the `@` property.
 
-				if (instr.getKey().matches(path)) {
+#### Resources
+Resources are targeted by using a glob pattern that ends with `*`.
```

Comment:
It is more correct to say not ending with `/`, right?

Reply:
sure.